pandas/core/frame.py

```diff
@@ -6422,19 +6424,68 @@ def sort_values(  # type: ignore[override]
         else:
             return result.__finalize__(self, method="sort_values")
 
-    @deprecate_nonkeyword_arguments(version=None, allowed_args=["self"])
+    @overload
+    def sort_index(
+        self,
+        *,
+        axis: Axis = ...,
+        level: Level | None = ...,
+        ascending: bool | int | Sequence[bool | int] = ...,
+        inplace: Literal[True],
+        kind: SortKind = ...,
+        na_position: NaPosition = ...,
+        sort_remaining: bool = ...,
+        ignore_index: bool = ...,
+        key: IndexKeyFunc = ...,
+    ) -> None:
+        ...
+
+    @overload
+    def sort_index(
+        self,
+        *,
+        axis: Axis = ...,
+        level: Level | None = ...,
+        ascending: bool | int | Sequence[bool | int] = ...,
+        inplace: Literal[False] = ...,
+        kind: SortKind = ...,
+        na_position: NaPosition = ...,
+        sort_remaining: bool = ...,
+        ignore_index: bool = ...,
+        key: IndexKeyFunc = ...,
+    ) -> DataFrame:
+        ...
+
+    @overload
     def sort_index(
+        self,
+        *,
+        axis: Axis = ...,
+        level: Level | None = ...,
+        ascending: bool | int | Sequence[bool | int] = ...,
+        inplace: bool = ...,
+        kind: SortKind = ...,
+        na_position: NaPosition = ...,
+        sort_remaining: bool = ...,
+        ignore_index: bool = ...,
+        key: IndexKeyFunc = ...,
+    ) -> DataFrame | None:
+        ...
+
+    # error: Signature of "sort_index" incompatible with supertype "NDFrame"
+    @deprecate_nonkeyword_arguments(version=None, allowed_args=["self"])
+    def sort_index(  # type: ignore[override]
```

Comment:
I think, but I'm not sure, that you can avoid the ignore by using `self: DataFrame` here, and also make sure that you are using the exact same argument types (e.g., for `sort_remaining` it is `bool_t` in `generic.py`, but `bool` here)

Reply:
frame had a missing return type and the sub-class behavior was for frame/series inconsistent with NDFrame - changed both, but still need this ignore.

generic.py needs (unfortunately bool_t - some limitation of mypy as NDFrame also has a bool method). I'd like to avoid the cryptic looking bool_t in other places. Using bool_t in frame, doesn't make a difference in terms of the ignore.

Reply:
> but still need this ignore.

looks similar to an issue that i've encountered before, I have a branch somewhere that I didn't open a PR for pending resolution of this.

the full error is 

```
pandas/core/frame.py:6484: error: Signature of "sort_index" incompatible with supertype "NDFrame"  [override]
pandas/core/frame.py:6484: note:      Superclass:
pandas/core/frame.py:6484: note:          @overload
pandas/core/frame.py:6484: note:          def sort_index(self, *, axis: Union[str, int] = ..., level: Optional[Union[Hashable, int]] = ..., ascending: Union[bool, Sequence[bool]] = ..., inplace: Literal[True], kind: Union[Literal['quicksort'], Literal['mergesort'], Literal['heapsort'], Literal['stable']] = ..., na_position: Union[Literal['first'], Literal['last']] = ..., sort_remaining: bool = ..., ignore_index: bool = ..., key: Optional[Callable[[Index], Union[Index, Union[Union[ExtensionArray, ndarray[Any, Any]], Index, Series]]]] = ...) -> None
pandas/core/frame.py:6484: note:          @overload
pandas/core/frame.py:6484: note:          def sort_index(self, *, axis: Union[str, int] = ..., level: Optional[Hashable] = ..., ascending: Union[bool, Sequence[bool]] = ..., inplace: Literal[False] = ..., kind: Union[Literal['quicksort'], Literal['mergesort'], Literal['heapsort'], Literal['stable']] = ..., na_position: Union[Literal['first'], Literal['last']] = ..., sort_remaining: bool = ..., ignore_index: bool = ..., key: Optional[Callable[[Index], Union[Index, Union[Union[ExtensionArray, ndarray[Any, Any]], Index, Series]]]] = ...) -> DataFrame
pandas/core/frame.py:6484: note:          @overload
pandas/core/frame.py:6484: note:          def sort_index(self, *, axis: Union[str, int] = ..., level: Optional[Hashable] = ..., ascending: Union[bool, Sequence[bool]] = ..., inplace: bool = ..., kind: Union[Literal['quicksort'], Literal['mergesort'], Literal['heapsort'], Literal['stable']] = ..., na_position: Union[Literal['first'], Literal['last']] = ..., sort_remaining: bool = ..., ignore_index: bool = ..., key: Optional[Callable[[Index], Union[Index, Union[Union[ExtensionArray, ndarray[Any, Any]], Index, Series]]]] = ...) -> Optional[DataFrame]
pandas/core/frame.py:6484: note:      Subclass:
pandas/core/frame.py:6484: note:          def sort_index(self, axis: Union[str, int] = ..., level: Optional[Hashable] = ..., ascending: Union[bool, Sequence[bool]] = ..., inplace: bool = ..., kind: Union[Literal['quicksort'], Literal['mergesort'], Literal['heapsort'], Literal['stable']] = ..., na_position: Union[Literal['first'], Literal['last']] = ..., sort_remaining: bool = ..., ignore_index: bool = ..., key: Optional[Callable[[Index], Union[Index, Union[Union[ExtensionArray, ndarray[Any, Any]], Index, Series]]]] = ...) -> Optional[DataFrame]
Found 1 error in 1 file (checked 1362 source files)
```

which does look like a false positive, since the message seems to imply that it is comparing the non-overload signature to the overload signature.

I'm ok with the ignore for now.

Reply:
I think it maybe down to not allowing positional arguments in the overloads